### ISSUE TITLE
[Support]: fatal: fetch-pack: invalid index-pack output 

### ISSUE THREAD

**Issue by tusharhero** [progfolio/elpaca#210]
### Confirmation

- [X] I have read the documentation and need clarification.
- [X] I have searched previous issues to see if my question is a duplicate.
- [ ] I am  checking these without reading them.
- [X] I have checked the Elpaca github wiki to see if my issue is covered.

### Elpaca Version

```
Elpaca ad60abc HEAD -> master, origin/master, origin/HEAD
installer:      0.6
emacs-version:  GNU Emacs 29.1 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.38, cairo version 1.17.8)
```

### Operating System

```
❯ uname -a
Linux archlinux 6.6.5-zen1-1-zen #1 ZEN SMP PREEMPT_DYNAMIC Fri, 08 Dec 2023 17:06:12 +0000 x86_64 GNU/Linux
```

### Description

```
rainbow-mode                   cloning              Cloning into '/home/tusharhero/.config/emacs/elpaca/repos/rainbow-mode'...       310.059665
vterm-toggle                   busy                 Vterm needs `vterm-module' to work.  Compile it now? (y or n)                    377.589352
breadcrumb                     failed               Failed dependencies: (project)                                                   546.392467
project                        failed               fatal: fetch-pack: invalid index-pack output                                     546.452853
eglot                          failed               fatal: fetch-pack: invalid index-pack output                                     549.406670
```
I am installing elpaca from my config which uses 0.6 installer, and has a bunch use package statements. I keep getting `fatal: fetch-pack: invalid index-pack output` specifically on projectel and eglot. Other packages seem to be fine.

I have tried deleting ~/.config/emacs and retrying but that didn't work.

**Comment by tusharhero** [comment]
I had a use-package call for project-el, which in inbuilt, I removed it and that seemed fixed the issues. Another thing I would like to point out is that elpaca was trying to clone the entirety of https://github.com/emacs-mirror/emacs .

Is that normal?

**Comment by progfolio** [comment]
> I keep getting fatal: fetch-pack: invalid index-pack output specifically on projectel and eglot.

If you want to use the builtin version of those packages, you can add `:elpaca nil` to your use-package declarations for them (I'm assuming you have `elpaca-use-package-by-default` is non-nil). e.g

```emacs-lisp
(use-package project :elpaca nil
  ;;etc
)
```

> Another thing I would like to point out is that elpaca was trying to clone the entirety of https://github.com/emacs-mirror/emacs .

> Is that normal?

If you want to install the development version of a builtin package, yes, you'll need a copy of the Emacs repository for core packages.
The default is a shallow clone (which can be adjusted via the `:depth` recipe keyword).

The fletch-pack error is due to an invalid object in GNU's repository. I'll see if I can add a workaround to prevent it. If memory serves, a shallow clone should work, though:

https://github.com/radian-software/straight.el/issues/994#issuecomment-1234922197

I'll have limited internet access for next few days, but will look into this when I can get to a computer. Thank you.

**Comment by tusharhero** [comment]
On Mon, Dec 11, 2023 at 01:22:03PM -0800, Nicholas Vollmer wrote:
> > I keep getting fatal: fetch-pack: invalid index-pack output specifically on projectel and eglot.
> 
> If you want to use the builtin version of those packages, you can add `:elpaca nil` to your use-package declarations for them (I'm assuming you have `elpaca-use-package-by-default` is non-nil). e.g
> 
> ```emacs-lisp
> (use-package project :elpaca nil
>   ;;etc
> )
> ```
> 
> > Another thing I would like to point out is that elpaca was trying to clone the entirety of https://github.com/emacs-mirror/emacs .
> 
> > Is that normal?
> 
> If you want to install the development version of a builtin package, yes, you'll need a copy of the Emacs repository for core packages.
> The default is a shallow clone (which can be adjusted via the `:depth` recipe keyword).
> 
> The fletch-pack error is due to an invalid object in GNU's repository. I'll see if I can add a workaround to prevent it. If memory serves, a shallow clone should work, though:
> 
> https://github.com/radian-software/straight.el/issues/994#issuecomment-1234922197
> 
> I'll have limited internet access for next few days, but will look into this when I can get to a computer. Thank you. 
> 
> -- 
> Reply to this email directly or view it on GitHub:
> https://github.com/progfolio/elpaca/issues/210#issuecomment-1850908294
> You are receiving this because you authored the thread.
> 
> Message ID: ***@***.***>

It used to work alright before, is this a new problem with their repository?
I have another question, why are we using github's mirror of the emacs repository, why not the GNU mirror?

---
tusharhero

**Comment by progfolio** [comment]
> It used to work alright before, is this a new problem with their repository?

No it's not a new issue. I can't speak to why it may have worked before for you.

> I have another question, why are we using github's mirror of the emacs repository, why not the GNU mirror?

Github tends to serve faster and more reliably than GNU's servers. The address used can be overridden via the  `:repo` recipe keyword if you prefer to clone from another server.

**Comment by progfolio** [comment]
Unfortunately, I'm unable to reproduce the issue on my end. See the following test case:


<details open><summary>Test Case</summary>

[How to run this test?](https://github.com/progfolio/elpaca/wiki/Troubleshooting#the-elpaca-test-macro)

```emacs-lisp
(elpaca-test
  :early-init (setq elpaca-menu-functions '(elpaca-menu-gnu-devel-elpa))
  :init (elpaca project)
  (elpaca-wait)
  (elpaca-test-log ".*"))
```

</details>
<details><summary>Host Env</summary>

<table>
<tr><td>elpaca</td><td>1008684 HEAD -> feat/version-checks</td>
<tr><td>isntaller</td><td>0.6</td>
<tr><td>emacs</td><td>GNU Emacs 30.0.50 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.38, cairo version 1.18.0) of 2023-12-06</td>
<tr><td>git</td><td>git version 2.43.0</td>
</table>
</details>

<details><summary>Output</summary>

```emacs-lisp
  INFO     Scraping files for loaddefs... 
  INFO     Scraping files for loaddefs...done
  GEN      ../elpaca-autoloads.el
Cloning into '/tmp/elpaca.26EvEJ/elpaca/repos/elpaca'...
Your branch is up to date with 'origin/master'.
Checking /tmp/elpaca.26EvEJ/elpaca/repos/elpaca...
Compiling /tmp/elpaca.26EvEJ/elpaca/repos/elpaca/elpaca-info.el...
Compiling /tmp/elpaca.26EvEJ/elpaca/repos/elpaca/elpaca-log.el...
Compiling /tmp/elpaca.26EvEJ/elpaca/repos/elpaca/elpaca-manager.el...
Compiling /tmp/elpaca.26EvEJ/elpaca/repos/elpaca/elpaca-menu-elpa.el...
Compiling /tmp/elpaca.26EvEJ/elpaca/repos/elpaca/elpaca-menu-melpa.el...
Compiling /tmp/elpaca.26EvEJ/elpaca/repos/elpaca/elpaca-menu-org.el...
Compiling /tmp/elpaca.26EvEJ/elpaca/repos/elpaca/elpaca-process.el...
Compiling /tmp/elpaca.26EvEJ/elpaca/repos/elpaca/elpaca-test.el...
Compiling /tmp/elpaca.26EvEJ/elpaca/repos/elpaca/elpaca-ui.el...
Compiling /tmp/elpaca.26EvEJ/elpaca/repos/elpaca/elpaca.el...
Checking /tmp/elpaca.26EvEJ/elpaca/repos/elpaca/doc...
Compiling /tmp/elpaca.26EvEJ/elpaca/repos/elpaca/doc/early-init.el...
Compiling /tmp/elpaca.26EvEJ/elpaca/repos/elpaca/doc/init.el...
Checking /tmp/elpaca.26EvEJ/elpaca/repos/elpaca/extensions...
Compiling /tmp/elpaca.26EvEJ/elpaca/repos/elpaca/extensions/elpaca-use-package.el...
Checking /tmp/elpaca.26EvEJ/elpaca/repos/elpaca/images...
Checking /tmp/elpaca.26EvEJ/elpaca/repos/elpaca/test...
Compiling /tmp/elpaca.26EvEJ/elpaca/repos/elpaca/test/elpaca-test.el...
Done (Total of 11 files compiled, 3 skipped in 4 directories)

Downloading GNU-devel ELPA...
elpaca-log: ".*"
"
elpaca                         unblocked            No external dependencies                                                         02.004240
elpaca                         linking              Linking build files                                                              02.004519
elpaca                         linking              Build files linked                                                               02.005885
elpaca                         autoloads            Generating autoloads: /tmp/elpaca.26EvEJ/elpaca/builds/elpaca                    02.006586
project                        cloning              Cloning                                                                          02.006896
elpaca                         autoloads              INFO     Scraping files for loaddefs...                                        02.107816
elpaca                         autoloads              INFO     Scraping files for loaddefs...done                                    02.118488
elpaca                         autoloads              GEN      elpaca-autoloads.el                                                   02.123150
elpaca                         autoloads            Autoloads Generated                                                              02.123556
elpaca                         byte-compilation     Byte compiling                                                                   02.123855
elpaca                         byte-compilation     Checking /tmp/elpaca.26EvEJ/elpaca/builds/elpaca...                              02.177761
elpaca                         byte-compilation     Compiling /tmp/elpaca.26EvEJ/elpaca/builds/elpaca/elpaca-autoloads.el...         02.181019
elpaca                         byte-compilation     Compiling /tmp/elpaca.26EvEJ/elpaca/builds/elpaca/elpaca-info.el...              02.181379
elpaca                         byte-compilation     Compiling /tmp/elpaca.26EvEJ/elpaca/builds/elpaca/elpaca-log.el...               02.259304
elpaca                         byte-compilation     Compiling /tmp/elpaca.26EvEJ/elpaca/builds/elpaca/elpaca-manager.el...           02.266771
elpaca                         byte-compilation     Compiling /tmp/elpaca.26EvEJ/elpaca/builds/elpaca/elpaca-menu-elpa.el...         02.272606
elpaca                         byte-compilation     Compiling /tmp/elpaca.26EvEJ/elpaca/builds/elpaca/elpaca-menu-melpa.el...        02.282691
elpaca                         byte-compilation     Compiling /tmp/elpaca.26EvEJ/elpaca/builds/elpaca/elpaca-menu-org.el...          02.283264
elpaca                         byte-compilation     Compiling /tmp/elpaca.26EvEJ/elpaca/builds/elpaca/elpaca-process.el...           02.283721
elpaca                         byte-compilation     Compiling /tmp/elpaca.26EvEJ/elpaca/builds/elpaca/elpaca-test.el...              02.284169
elpaca                         byte-compilation     Compiling /tmp/elpaca.26EvEJ/elpaca/builds/elpaca/elpaca-ui.el...                02.292772
elpaca                         byte-compilation     Compiling /tmp/elpaca.26EvEJ/elpaca/builds/elpaca/elpaca.el...                   02.306209
elpaca                         byte-compilation     Done (Total of 10 files compiled, 1 skipped)                                     02.375527
elpaca                         byte-compilation     Byte compilation complete                                                        02.379647
elpaca                         info                 Compiling Info files                                                             02.380068
elpaca                         info                 Info compiled                                                                    02.563314
elpaca                         info                 Installing Info files                                                            02.563796
elpaca                         info                 /tmp/elpaca.26EvEJ/elpaca/builds/elpaca/elpaca.info                              02.564555
elpaca                         info                 Info installed                                                                   02.566051
elpaca                         info                 Adding Info path                                                                 02.566478
elpaca                         finished             ✓ 0.575 secs                                                                     02.566853
project                        cloning              Resolving deltas: 100% (645/645), done.                                          21.415590
project                        checking-out-ref     Checking out master                                                              22.127610
project                        checking-out-ref     Reset branch 'master'                                                            22.279170
project                        checking-out-ref     branch 'master' set up to track 'origin/master'.                                 22.279871
project                        checking-out-ref     Your branch is up to date with 'origin/master'.                                  22.280387
project                        ref-checked-out      master checked out                                                               22.280943
project                        unblocked            No external dependencies                                                         22.418709
project                        linking              Linking build files                                                              22.419157
project                        linking              Build files linked                                                               22.419659
project                        autoloads            Generating autoloads: /tmp/elpaca.26EvEJ/elpaca/builds/project                   22.420212
project                        autoloads              INFO     Scraping files for loaddefs...                                        22.525502
project                        autoloads              INFO     Scraping files for loaddefs...done                                    22.532495
project                        autoloads              GEN      project-autoloads.el                                                  22.535911
project                        autoloads            Autoloads Generated                                                              22.536339
project                        byte-compilation     Byte compiling                                                                   22.536684
project                        byte-compilation     Checking /tmp/elpaca.26EvEJ/elpaca/builds/project...                             22.597847
project                        byte-compilation     Compiling /tmp/elpaca.26EvEJ/elpaca/builds/project/project-autoloads.el...       22.604834
project                        byte-compilation     Compiling /tmp/elpaca.26EvEJ/elpaca/builds/project/project.el...                 22.605399
project                        byte-compilation     Done (Total of 1 file compiled, 1 skipped)                                       22.651650
project                        byte-compilation     Byte compilation complete                                                        22.655135
project                        info                 Compiling Info files                                                             22.655567
project                        info                 Info source files not found                                                      22.655934
project                        activation           Activating package                                                               22.656289
project                        activation           Package build dir added to load-path                                             22.656629
project                        activation           Caching autoloads                                                                22.656961
project                        activation           Autoloads cached                                                                 22.657485
project                        finished             ✓ 20.666 secs                                                                    22.657854
"

 Test Env

Elpaca ad60abc HEAD -> master, origin/master, origin/HEAD
installer:      0.6
emacs-version:  GNU Emacs 30.0.50 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.38, cairo version 1.18.0)
 of 2023-12-06
git --version:  git version 2.43.0
```
</details>


Perhaps it is/was a networking issue?

**Comment by tusharhero** [comment]
I haven't had this problem since. I am closing this issue.

**Comment by progfolio** [comment]
Likely a temporary networking hiccup, then. Thanks for letting me know.
